metrics/src/counter.rs

```diff
@@ -95,6 +95,10 @@ impl Counter {
         );
     }
     pub fn inc(&mut self, level: log::Level, events: usize) {
+        if !log_enabled!(level) {
```

Comment:
this will drop the metrics if the log level is not enabled.

Reply:
@pgarg66 Logging is by both level and module name, looks like this call would only take into account the counter/metrics module and not the callers.

Reply:
https://doc.rust-lang.org/1.1.0/log/macro.log_enabled!.html implies that `module_path!()` is used in its implementation, is that not really the case?

Reply:
What is the module path at this point, `solana-metrics` or `solana`?  Also, just module path is too broad, filtering by filename or metric name would be more helpful since pretty much all the metrics are in `solana`.  Filtering by level imo is fundamentally flawed because if you want to turn on more detail you get the world which in itself is not scalable.  The more detailed you want to get the more you want to focus on a particular subset of metrics without having to wade through or be encumbered by all the rest of the metrics.

Reply:
Same semantics as logging right?   Module is the .rs file.

Reply:
Almost.   In this case, this will always resolve to `solana-metrics::counter`?

Reply:
If it worked like that then RUST_LOG would never behave as it does.

Reply:
`log` calls `module_path!` in the macro, here we are not.  Maybe I'm missing something...

Reply:
Or maybe I am.  Did you confirm it doesn't work as I think it does? :)

Reply:
I'm getting a little confused, not sure how you think it does ;-)

From my understanding, `module_path!` returns the module name.   For example, if you have a package named `package` and in its source directory you have main.rs and other.rs then `module_path!` called from those files returns respectively:

`package`
`package::other`

But if you call `module_path!` in a macro it returns the module path of where the macro was called.

Reply:
I've confirmed the above

Reply:
weird, ok.  well that's a bug

Reply:
What's a bug?

Reply:
@pgarg66 

Can you move the `log_enabled!` call up to the macro level so that the right module path is used for filtering and so filtering occurs up front?

Reply:
👍